server/pbench/bin/pbench-server-prep-shim-001

```diff
@@ -0,0 +1,220 @@
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+. $dir/pbench-base.sh
+###########################################################################
+
+archive=$(getconf.py archive-dir pbench-server)
+quarantine=$(getconf.py quarantine-dir pbench-server)
+duplicates=$(getconf.py duplicates-dir pbench-server)
+
+errlog=$LOGSDIR/$PROG/$PROG.error
+if [[ "$archive" == "" ]] ;then
+    echo "Failed: \"getconf.py archive-dir pbench-server" >> $errlog
+    exit 1
+fi
+
+if [[ "$quarantine" == "" ]] ;then
+    echo "Failed: \"getconf.py quarantine-dir pbench-server" $errlog
+    exit 1
+fi
+
+if [[ "$duplicates" == "" ]] ;then
+    echo "Failed: \"getconf.py duplicates-dir pbench-server" $errlog
+    exit 1
+fi
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+tmp=$TMP/$PROG.$$
+
+function quarantine () {
+    dest=$1
+    shift
+    files="$@"
+
+    mkdir -p $dest || return $?
+    mv $files $dest || return $?
+}
+
+trap 'rm -rf $tmp' EXIT INT QUIT
+
+mkdir -p $tmp
+list=$tmp/list.tb
+status=$tmp/status
+> $status
+# we are explicitly handling version-001 data
+results_dir=$(getconf.py receive-dir-${version} pbench-server)
+
+# check for results that are ready for processing: version 001
+# agents create a link in the TODO subdir (which is created if
+# necessary) after they finish with MD5 checking, so that's what
+# we look for.
+find ${results_dir} -maxdepth 3 -type l -name '*.tar.xz' > $list
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
+while read tblink ;do
+    # resolve the link
+    tb=$(readlink -f ${tblink})
+    sts=$?
+    if [ $sts != 0 ] ;then
+        echo "$TS: readlink failed: ${tblink}" | tee -a $status >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # hostname is the last component of the directory part of the full path
+    controller=$(basename ${tbdir})
+
+    # TBD
+    dest=$archive/$controller
+
+    # sanity checks
+    # Impossible: the tarball disappeared between the find and this check!
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: ${tb} file does not exist" | tee -a $status >&4
+        nerrs=$nerrs+1
+        continue
+    fi
+
+    if [ ! -f ${tb}.md5 ] ;then
+        # this may be the result of pbench-move-results racing with this script:
+        # we'll just skip the tarball this time and try again next time.
+        echo "$TS: ${tb}.md5 file does not exist" >&4
+        continue
+    fi
+
+    # This should not happen: version 001 agents check for duplicates in pbench-move-results.
```

Comment:
Perhaps reword this comment to say, "While version 001 agents check for duplicates in their copy of `pbench-move-results`, they are only able to check against their 'archive` location, not against the real archive location.  So this check is still necessary to avoid actual duplicates."